docs/contribute/plugin/best_practice.md

```diff
@@ -0,0 +1,77 @@
+   
+## Writing and Compiling the Plugin
+
+To create a plugin, you need to define a function in C or C++, and declare it with the `extern "C"` keyword to ensure that the function is exported using C-compatible naming conventions. Then, compile the function as a shared library.
+
+## Registering the Plugin with WasmEdge Runtime
+
+After compiling the shared library, you need to register the function with the WasmEdge runtime. 
+
+## Testing the Plugin
+
+Testing is a crucial part of the plugin development process. It ensures that the plugin behaves as expected, meets the requirements, and performs optimally. WasmEdge provides a set of tests for various plugins in its [repository](https://github.com/WasmEdge/WasmEdge/tree/master/test/plugins) that you can use as references for writing your own tests.
+
+To run tests for the WasmEdge plugin, you'll need to follow a few steps. In this case, we'll use the `wasmedge-image` plugin as an example.
+
+   - **Step 1: Build the WasmEdge Runtime and WasmEdge-image Plugin** 
+   First, you need to build both the WasmEdge runtime and the wasmedge-image plugin. To do this, you can clone the WasmEdge repository and build it using the following commands:
+
+   ```bash
+   git clone https://github.com/WasmEdge/WasmEdge.git
+   cd WasmEdge
+   mkdir build && cd build
+   cmake ..
+   make
+   sudo make install
+   ```
+
+   You also need to build the `wasmedge-image` plugin. Use the following command:
+
+   ```bash
+   cd ../plugins/wasmedge_image
+   mkdir build && cd build
+   cmake ..
+   make
+   sudo make install
+   ```
+
+   - **Step 2: Run the Tests**
+   The WasmEdge repository provides a set of tests for various plugins, including `wasmedge-image`. You can find the test cases in the `test/plugins/wasmedge_image` directory of the repository.
+
+      To run these tests, you can use the `ctest` command from the build directory:
+
+      ```bash
+      cd ../../../test/plugins/wasmedge_image
+      mkdir build && cd build
+      cmake ..
+      make
+      ctest
+      ```
+
+   This will run all the unit tests and integration tests for the `wasmedge-image `plugin. These tests ensure that the plugin behaves as expected, meets the requirements, and performs optimally. They also verify that the plugin correctly integrates with the WebAssembly program and that the WebAssembly program can call the plugin's functions correctly.
+
+   - **Step 3: Analyze the Test Results**
+   After running the tests, analyze the results to identify any issues or bugs. If any test fails, you should debug the issue, fix the problem, and then rerun the tests to ensure that the fix works as expected.
+
+By following these steps, you can effectively run tests for the `wasmedge-image` plugin or any other WasmEdge plugin. Also if you want to develop your own tests follow [Writing Tests for WasmEdge Plugins](test_plugin.md) for details.
+
+## Publishing the Plugin
```

Comment:
How to publish? If i want to contribute my code to the WasmEdge repo, what should I do?